src/Aspire.Hosting/Dashboard/ResourceCollection.cs

```diff
@@ -8,26 +8,16 @@
 namespace Aspire.Hosting.Dashboard;
 
 /// <summary>
-/// Builds a collection of resources by integrating incoming changes from a channel,
-/// and allowing multiple subscribers to receive the current resource snapshot and future
-/// updates.
+/// Builds a collection of resources by integrating incoming resource changes,
+/// and allowing multiple subscribers to receive the current resource collection
+/// snapshot and future updates.
 /// </summary>
-internal sealed class ResourceCollection
+internal sealed class ResourceCollection(CancellationToken cancellationToken)
 {
     private readonly object _syncLock = new();
-    private readonly Channel<ResourceChange> _incomingChannel;
-    private readonly CancellationToken _cancellationToken;
     private readonly Dictionary<string, ResourceViewModel> _snapshot = [];
     private ImmutableHashSet<Channel<ResourceChange>> _outgoingChannels = [];
 
-    public ResourceCollection(Channel<ResourceChange> incomingChannel, CancellationToken cancellationToken)
-    {
-        _incomingChannel = incomingChannel;
-        _cancellationToken = cancellationToken;
-
-        Task.Run(ProcessChanges, cancellationToken);
-    }
-
     public ResourceSubscription Subscribe()
```

Comment:
Currently the subscribers are the various user's in the dashboard UI (resources, console logs, peer resolver). After the move to gRPC, will it change to each gRPC call being a subscriber?

Reply:
No there'll be a single client subscription that fans out to interested parties within the dashboard. You can see more detail in the message I shared on Teams. My current view of the end state is:

![image](https://github.com/dotnet/aspire/assets/350947/5f7a5b54-a792-4cd5-9418-d0649f18b81c)

There's some detail on Teams. If you want to discuss this aspect further -- given it's further out -- please add your thoughts on that thread rather than here.

Reply:
End state updated slightly on the client side:

![image](https://github.com/dotnet/aspire/assets/350947/a94a746a-9d9f-4712-9e5d-5ac80502e718)